topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.md

```diff
@@ -0,0 +1,532 @@
+>         - *"Annotated data matrix to add"*: `Select all other Manipulate AnnData (transpose) outputs`
+>         - *"Join method"*: `Intersection of variables`
+>         - *"Key to add the batch annotation to obs"*: `batch`
+>         - *"Separator to join the existing index names with the batch category"*: `-`
+>
+> 2. **Rename** {% icon galaxy-pencil %} output `Combined Object`
+>
+{: .hands_on}
+
+Now we have all of our gene expression data stored in a single AnnData object! Now we just need to add our metadata!
+
+# Annotating the Data
+
+The next step is to annotate our data using the information gathered from the excel sheet earlier, we will do this by leveraging the batch column generated when the objects were combined for each individual patient/sample (indicated by the batch number) we will add in the relevant annotations. First lets extract the observation data so we can manipulate it.
+
+> <hands-on-title></hands-on-title>
+>
+> 1. {% tool [Inspect AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Combined Object`
+>    - *"What to inspect?"*: `Key-indexed observations annotation (obs)`
+>
+> 2. **Rename** {% icon galaxy-pencil %} output `Observation data`
+>
+{: .hands_on}
+
+Now for each new column we want to add we need to replace the relvant batch numbers with the appropriate metadata. After that we can cut the modified column out so it is separated. We will repeat this process for each piece of metadata we want to add, all the relevant data can be found in the first table of this tutorial.
+
+Lets now add the replicate column which tells us which rows are part of pools of the same patient and tumor location.
+
+> <hands-on-title>Create replicate metadata</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `Observation data`
+>    - *"1: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `2|4|7|8`
+>         - *"Replace with"*: `poolA`
+>    - **+ Insert Replacement**
+>    - *"2: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `3|5|9`
+>         - *"Replace with"*: `poolB`
+>    - **+ Insert Replacement**
+>    - *"3: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `6`
+>         - *"Replace with"*: `poolC`
+>    - **+ Insert Replacement**
+>    - *"4: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `0|1`
+>         - *"Replace with"*: `NA`
+>    - **+ Insert Replacement**
+>    - *"5: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `batch`
+>         - *"Replace with"*: `replicate`
+>
+> 2. {% tool [Cut](Cut1) %} with the following parameters:
```

Comment:
I suspect you could speed this section up by removing the 'Cut' option and using the Multi-Join
(combine multiple files)
(Galaxy Version 1.1.1) instead, using the initial barcode column as the key, and THEN cutting once to remove it